Marlin/src/gcode/calibrate/G28.cpp

```diff
@@ -411,11 +411,18 @@ void GcodeSuite::G28() {
     TERN_(HOME_Z_FIRST, if (doZ) homeaxis(Z_AXIS));
 
     const bool seenR = parser.seenval('R');
-    const float z_homing_height = seenR ? parser.value_linear_units() : Z_HOMING_HEIGHT;
+    float z_homing_height;
+    if (seenR)
+      z_homing_height = current_position.z + parser.value_linear_units();
+    else {
+      z_homing_height = (Z_CLEARANCE_BETWEEN_PROBES ) + _MIN(probe.offset.z, 0);
```

Comment:
It's better to use `Z_HOMING_HEIGHT` here within `G28`. It will be automatically set to `Z_CLEARANCE_BETWEEN_PROBES` if it is not provided. Considering the case where the BLTouch in HS mode is deployed while an XY move is being done, there should only be extra clearance after the probe has been deployed, and it may not be needed for the initial move to home X and/or Y, unless there is some logic to deploy the Z probe at the very start of `G28` after the initial Z raise. AFAIK the BLTouch doesn't ever deploy until it actually reaches the safe homing position, regardless of its HS mode.